homeassistant/components/nexia/util.py

```diff
@@ -16,3 +23,64 @@ def percent_conv(val):
     if val is None:
         return None
     return round(val * 100.0, 1)
+
+
+class SingleShot:
+    """Provides a single shot timer that can be reset.
+
+    Fires a while following the *last* call to `reset_delayed_action_trigger`.
+    Caller should use `async_shutdown` to clean up when done.
+    """
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        delay: timedelta,
+        delayed_call: Callable[[], Coroutine[Any, Any, None]],
+    ) -> None:
+        """Initialize this single shot timer."""
+        self._hass = hass
+        self._delay = delay
+        self._delayed_call = delayed_call
+        self._cancel_delayed_action: Callable[[], None] | None = None
+        self._execute_lock = asyncio.Lock()
+        self._shutting_down = False
+        self._job = HassJob(
+            self._delayed_action,
+            f"resettable single shot action {self._delay}",
+        )
```

Comment:
I think HassJob can be avoided here since we always know which type of callable `delayed_call` will be